server/sonar-server/src/test/java/org/sonar/server/test/ws/ListActionTest.java

```diff
@@ -31,280 +31,255 @@
+    insertTests(testFile, test);
+
+    ListResponse request = call(ws.newRequest().setParam(TEST_ID, test.getUuid()));
+
+    assertThat(request.getTestsList()).hasSize(1);
+    WsTests.Test result = request.getTests(0);
+    assertThat(result.getId()).isEqualTo(test.getUuid());
+    assertThat(result.getName()).isEqualTo(test.getName());
+    assertThat(result.getStatus()).isEqualTo(WsTests.TestStatus.OK);
+    assertThat(result.getFileId()).isEqualTo(testFile.uuid());
+    assertThat(result.getFileKey()).isEqualTo(testFile.key());
+    assertThat(result.getFileName()).isEqualTo(testFile.path());
+    assertThat(result.getDurationInMs()).isEqualTo(test.getExecutionTimeMs());
+    assertThat(result.getMessage()).isEqualTo(test.getMsg());
+    assertThat(result.getStacktrace()).isEqualTo(test.getStacktrace());
+    assertThat(result.getCoveredLines()).isEqualTo(6);
+  }
 
-    es.putDocuments(TestIndexDefinition.INDEX, TestIndexDefinition.TYPE, TestFile1.doc());
+  @Test
+  public void return_tests_based_on_test_uuid() throws Exception {
+    userSessionRule.addProjectUuidPermissions(CODEVIEWER, project.uuid());
+    DbFileSources.Test test1 = newTest(mainFile, 10).build();
+    DbFileSources.Test test2 = newTest(mainFile, 11).build();
+    insertTests(testFile, test1, test2);
 
-    WsTester.TestRequest request = ws.newGetRequest("api/tests", "list").setParam(ListAction.TEST_ID, TestFile1.UUID);
+    ListResponse request = call(ws.newRequest().setParam(TEST_ID, test1.getUuid()));
 
-    request.execute().assertJson(getClass(), "list-test-uuid.json");
+    assertThat(request.getTestsList()).extracting(WsTests.Test::getId).containsOnly(test1.getUuid());
   }
 
   @Test
-  public void list_based_on_test_file_uuid() throws Exception {
-    userSessionRule.addProjectUuidPermissions(UserRole.CODEVIEWER, TestFile1.PROJECT_UUID);
-    dbClient.componentDao().insert(db.getSession(), TestFile1.dto());
-    db.getSession().commit();
-
-    es.putDocuments(TestIndexDefinition.INDEX, TestIndexDefinition.TYPE, TestFile1.doc());
+  public void return_tests_based_on_test_file_uuid() throws Exception {
+    userSessionRule.addProjectUuidPermissions(CODEVIEWER, project.uuid());
+    ComponentDto anotherTestFile = db.components().insertComponent(newFileDto(project));
+    DbFileSources.Test test1 = newTest(mainFile, 10).build();
+    DbFileSources.Test test2 = newTest(mainFile, 11).build();
+    DbFileSources.Test test3 = newTest(mainFile, 12).build();
+    insertTests(testFile, test1, test2);
+    insertTests(anotherTestFile, test3);
+
+    ListResponse request = call(ws.newRequest().setParam(TEST_FILE_ID, testFile.uuid()));
+
+    assertThat(request.getTestsList()).extracting(WsTests.Test::getId).containsOnly(test1.getUuid(), test2.getUuid());
+  }
 
-    WsTester.TestRequest request = ws.newGetRequest("api/tests", "list").setParam(ListAction.TEST_FILE_ID, TestFile1.FILE_UUID);
+  @Test
+  public void return_tests_based_on_test_file_key() throws Exception {
```

Comment:
I don't know, I'm trying to find the best sentence that is describing what is tested, but it's not easy for me